tests/common/plugins/sanity_check/__init__.py

```diff
@@ -79,18 +79,36 @@ def _update_check_items(old_items, new_items, supported_items):
 
 def print_logs(duthosts):
     for dut in duthosts:
-        logger.info("Run commands to print logs, logs to be collected on {}:\n{}"\
-            .format(dut.hostname, json.dumps(constants.PRINT_LOGS, indent=4)))
-        for cmd in constants.PRINT_LOGS.values():
-            res = dut.shell(cmd, module_ignore_errors=True, verbose=False)
-            logger.info("cmd='%s', output:\n%s" % (cmd, json.dumps(res["stdout_lines"], indent=4)))
+        logger.info("Run commands to print logs")
+
+        cmds = constants.PRINT_LOGS.values()
+        results = dut.shell_cmds(cmds=cmds, module_ignore_errors=True, verbose=False)['results']
+        outputs = []
+        for res in results:
+            res.pop('stdout')
+            res.pop('stderr')
+            outputs.append(res)
+        logger.info(json.dumps(outputs, indent=4))
 
 
-def do_checks(request, check_items):
+def filter_check_items(tbinfo, check_items):
+    filtered_check_items = copy.deepcopy(check_items)
+
+    # ignore BGP check for particular topology type
+    if tbinfo['topo']['type'] == 'ptf' and 'bgp' in filtered_check_items:
+        filtered_check_items.remove('bgp')
+
+    if 'dualtor' not in tbinfo['topo']['name'] and 'mux_simulator' in filtered_check_items:
+        filtered_check_items.remove('mux_simulator')
+
+    return filtered_check_items
+
+
+def do_checks(request, check_items, *args, **kwargs):
     check_results = []
     for item in check_items:
         check_fixture = request.getfixturevalue(_item2fixture(item))
-        results = check_fixture()
+        results = check_fixture(*args, **kwargs)
```

Comment:
Hi Xin, can we add the request in parameters? So we have the context info in the sanity check, so we can save the info per module name.

Reply:
When you define a new check, the check must be a fixture that returns a function that is responsible for the actual checking. The `request` object itself is a pytest fixture. So, you can get the `request` object just in your check fixture by specifying it in the argument list. For example:
```
@pytest.fixture(scope='module')
def check_something(request):
    def _check(*args, **kwargs):
        # Run checking
        # The request object is available in this scope.
        return check_result
    return _check
```